File: rag-service/rag_service/llm/embeddings.py
from __future__ import annotations

from rag_service.contracts import AppSettingsPayload
from rag_service.llm import ollama, openai_compatible


def embed_text(
    settings: AppSettingsPayload,
    api_key: str | None,
    text: str,
) -> list[float]:
    if settings.active_provider == "ollama":
        return ollama.ollama_embed(
            settings.ollama.host,
            settings.ollama.embed_model,
            text,
        )
    return openai_compatible.openai_compatible_embed(
        settings.custom_http.base_url,
        api_key,
        settings.custom_http.embed_model,
        text,
        settings.rag.embedding_dimensions,
    )


def complete_chat(
    settings: AppSettingsPayload,
    api_key: str | None,
    messages: list[dict[str, str]],
) -> str:
    if settings.active_provider == "ollama":
        return ollama.ollama_chat(
            settings.ollama.host,
            settings.ollama.chat_model,
            messages,
        )
    return openai_compatible.openai_compatible_chat(
        settings.custom_http.base_url,
        api_key,
        settings.custom_http.model,
        messages,
    )

File: packages/llm/src/embeddings.ts
import type { AppSettings } from "@localrag/config";
import { ollamaEmbed } from "./ollama.js";
import { openAiCompatibleEmbed } from "./openai-compatible.js";

export async function embedText(
  settings: AppSettings,
  apiKey: string | null,
  text: string,
): Promise<number[]> {
  if (settings.activeProvider === "ollama") {
    return ollamaEmbed(
      settings.ollama.host,
      settings.ollama.embedModel,
      text,
    );
  }
  return openAiCompatibleEmbed(
    settings.customHttp.baseUrl,
    apiKey,
    settings.customHttp.embedModel,
    text,
    settings.rag.embeddingDimensions,
  );
}
